src/test/ts_tests/runSmokeTest.ts

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+/* eslint-disable @typescript-eslint/naming-convention */
+
+import * as cp from 'child_process';
+import * as path from 'path';
+
+import { runTests, downloadAndUnzipVSCode, resolveCliArgsFromVSCodeExecutablePath } from '@vscode/test-electron';
+import { EXTENSION_ROOT_DIR } from '../../common/constants';
+
+const TEST_PROJECT_DIR = path.join(EXTENSION_ROOT_DIR, 'src', 'test', 'ts_tests', 'test_data', 'project');
+
+async function main() {
+    try {
+        const vscodeExecutablePath = await downloadAndUnzipVSCode('stable');
+
+        const [cli, ...args] = resolveCliArgsFromVSCodeExecutablePath(vscodeExecutablePath);
```

Comment:
What is the purpose of cli and args from VS Code path?

Reply:
The `vscode-test` package downloads and installs a portable version of VS Code for testing. That VS Code has its own extension folder and user data folder so that it does not destroy your regular VS code install. The `cli` there is the path to VS Code executable, and the `args` there are `--extension-dir=<extension path>` and `--user-data=<user data>`. This is where any required extension like python and pylance are installed for smoke testing Black extension.